### PR TITLE
Unhardcode Gills mutations

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
Move mutations into json

#### Describe the solution
Moved gills mutations into a json flag.

### CODE REVIEW

File: doc/JSON_FLAGS.md [hunk 1]
```diff
@@ -1537,3 +1537,4 @@ Gun fault flags:
 - ```ALARMCLOCK``` You always can set alarms.
 - ```PARAIMMUNE``` You are immune to parasites.
 - ```IMMUNE_SPOIL``` You are immune to negative outcomes from spoiled food.
+- ```GILLS``` You can breath underwater.
```
Comment (actual-nh) [inline doc/JSON_FLAGS.md:1540]:
```suggestion
- ```GILLS``` You can breathe underwater.
```

File: src/character.cpp [hunk 2]
```diff
@@ -8760,19 +8759,6 @@ void Character::update_stamina( int turns )
     }
 
     const int max_stam = get_stamina_max();
-    if( get_power_level() >= 3_kJ && has_active_bionic( bio_gills ) ) {
-        int bonus = std::min<int>( units::to_kilojoule( get_power_level() ) / 3,
-                                   max_stam - get_stamina() - stamina_recovery * turns );
-        // so the effective recovery is up to 5x default
```
Comment (actual-nh) [inline src/character.cpp:8766]: The bonus and power consumption are no longer variable?

File: data/json/bionics.json [hunk 3]
```diff
@@ -529,9 +529,13 @@
     "id": "bio_gills",
     "type": "bionic",
     "name": { "str": "Respirator" },
-    "description": "A complex respiration augmentation system.  Improves respiration ability in air and allows breathing water.  Will automatically turn on when drowning.  Turn on to recharge stamina faster, at moderate power cost.  Asthmatics may also use it to stop asthma attacks.",
+    "description": "A complex respiration augmentation system.  Improves respiration ability in air and allows breathing water.  Turn on to recharge stamina faster, at moderate power cost.  Asthmatics may also use it to stop asthma attacks.",
     "occupied_bodyparts": [ [ "torso", 8 ], [ "head", 2 ], [ "mouth", 2 ] ],
-    "flags": [ "BIONIC_TOGGLED" ]
+    "flags": [ "BIONIC_TOGGLED" ],
+    "active_flags": [ "GILLS" ],
+    "react_cost": "5 kJ",
```
Comment (actual-nh) [inline data/json/bionics.json:536]:
```suggestion
    "react_cost": "3 kJ",
```
The previous minimum was 3 kJ; why the change?

**Review by actual-nh — COMMENTED** [review]
(What are you meaning in the PR by "will use stamina"?) The asthma portion could be handled with an effect that cancels it, or possibly the enchantment itself canceling it (I don't know if the latter is possible currently).

File: data/json/bionics.json (hunk 3 above)
Comment (actual-nh) [inline data/json/bionics.json:536]:
```suggestion
    "react_cost": "4 kJ",
```
By the default 20 base stamina regeneration rate, *3 gives you 40 more per turn, which would previously have cost 4kJ.

**Comment by Ramza13** [comment]
So will use stamina was a typo and I just fixed it to will use power.  As for power usage, the way it used to work was scaled by how much it regened.  Rather than try to preserve that somehow in json I opted to just make it a flat cost, since its always using power now I decided to make the cost lower than it had been to make up for the fact its always using it.  

I'm not married to the numbers and to some extent hope at some point in 0.G to break this apart into separate bionics.  Given the framework we have for enchantments and bionics I don't see a good way to preserve the variable cost system and made a judgement call on dropping it.  I'm not 100% happy with changing how it works and am open to suggestions but at the moment don't see a way outside of a massive overhaul of enchantments to unhardcode this without some changes.

**Comment by actual-nh** [comment]
> So will use stamina was a typo and I just fixed it to will use power. As for power usage, the way it used to work was scaled by how much it regened. Rather than try to preserve that somehow in json I opted to just make it a flat cost, since its always using power now I decided to make the cost lower than it had been to make up for the fact its always using it.

Understand on the typo. How is 5 kJ lower? Did I misunderstand something in the original code?

> I'm not married to the numbers and hope at some point in 0.G to break this apart into separate bionics as part of a bigger effort to change the bionics. Given the framework we have for enchantments and bionics I don't see a good way to preserve the variable cost system and made a judgement call on dropping it. I'm not 100% happy with changing how it works and am open to suggestions but at the moment don't see a way outside of a massive overhaul of enchantments to unhardcode this without some changes.

Understand! It's also involved with that a lot of the bionics would make more sense if they were self-triggered.

**Comment by Ramza13** [comment]
> 
> 
> > So will use stamina was a typo and I just fixed it to will use power. As for power usage, the way it used to work was scaled by how much it regened. Rather than try to preserve that somehow in json I opted to just make it a flat cost, since its always using power now I decided to make the cost lower than it had been to make up for the fact its always using it.
> 
> Understand on the typo. How is 5 kJ lower? Did I misunderstand something in the original code?
> 
> > I'm not married to the numbers and hope at some point in 0.G to break this apart into separate bionics as part of a bigger effort to change the bionics. Given the framework we have for enchantments and bionics I don't see a good way to preserve the variable cost system and made a judgement call on dropping it. I'm not 100% happy with changing how it works and am open to suggestions but at the moment don't see a way outside of a massive overhaul of enchantments to unhardcode this without some changes.
> 
> Understand! It's also involved with that a lot of the bionics would make more sense if they were self-triggered.

Whoops thought I made that 3kJ instead.  I'll make that change.  

Hmm a generic json system of triggers for bionics activating and deactivating sounds like a good idea.  I could probably do it by adding a few more dialog effects/conditions into into #42786, so I will wait until that goes in, I'm making a note of it though.

**Comment by Fris0uman** [comment]
Could you move the GILLS flag to a spell instead? It would help unifying stuffs

**Comment by Ramza13** [comment]
> 
> 
> Could you move the GILLS flag to a spell instead? It would help unifying stuffs

I'll do you one better.  #47633 now means all character/bionic flags can be used on effects and vice versa which can be added by spells.

**Comment by Fris0uman** [comment]
Very nice!
